### ISSUE TITLE
astrojs/db doesn't work on cloudflare pages

### ISSUE THREAD

**Issue by zeshhaan** [withastro/astro#12019]
### Astro Info

```block
$ astro info
Astro                    v5.0.0-beta.1
Node                     v20.10.0
System                   macOS (arm64)
Package Manager          bun
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/react
                         @astrojs/tailwind
                         astro:db
```

### Describe the Bug

@astrojs/db when used with [libsql credentials](https://docs.astro.build/en/guides/astro-db/#libsql) doesn't work on cloudflare pages. it works fine on local but when accessing a page with astro db query on preview/production env, it breaks with `500`. The console gives the following error message: 
`LibsqlError: SERVER_ERROR: Server returned HTTP status 401`

### What's the expected result?

When interacting with Astro DB using libsql credentials in a Cloudflare deployed site, say for example, querying a db, we're expected to see it return a result. But instead, it breaks and returns 500 error. 

### Link to Minimal Reproducible Example

https://github.com/zeshhaan/error-astrodb-cf-repro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by atej** [comment]
Running into the same issue trying to use `astro-db` with Turso on a Cloudflare Pages deployment.

**Comment by ematipico** [comment]
Is anyone able to provide better logs here?

**Comment by zeshhaan** [comment]
> Is anyone able to provide better logs here?

there was a discussion at discord with folks sharing the logs (if that might help)
https://discord.com/channels/830184174198718474/1285132367929020416

**Comment by misaelvillaverde** [comment]
Same issue, I'm getting:

```json
{
    "code": "SERVER_ERROR",
    "name": "LibsqlError"
}
```

Works fine locally (and astro dev --remote). Don't work when on cloudflare workers environment, i.e. wrangler dev or deployed to cloudflare pages. Also, `compatibility_flags = ["nodejs_compat"]`, don't change anything. Can't get a meaningful error for the time being.

**Comment by nootsponge** [comment]
Hello! Looks like the issue is to do with how the database variable is instantiated. Stack traces aren't entirely useful since it's all compiled and random looking but through some thorough sniffing around I found out that:

1) The parameters for `remoteUrl` and `appToken` are being assigned to variables that do not exist; and
2) Workers use a passed variable in the `fetch()` function to give the application its correct environment, NOT `process.env` like you would assume.

![image](https://github.com/user-attachments/assets/12530c31-301b-4e7e-99d1-6121ae25be6f)

So, in summary, Workers only exposes environment variables per `fetch()` as the passed `env` parameter, which I cannot find how to access via the generated page module.

I'd love to contribute to fix this if anyone can point me in the right direction.

**Comment by nathanpower** [comment]
> I'd love to contribute to fix this if anyone can point me in the right direction.

Not sure about the worker runtime, but I just hit a similar issue because I was on Node 20 and support for `.env` loading is experimental. I had to run with `node --env-file=.env ./dist/server/entry.mjs`

**Comment by haflinger** [comment]
> Hello! Looks like the issue is to do with how the database variable is instantiated. Stack traces aren't entirely useful since it's all compiled and random looking but through some thorough sniffing around I found out that:
> 
> 1. The parameters for `remoteUrl` and `appToken` are being assigned to variables that do not exist; and
> 2. Workers use a passed variable in the `fetch()` function to give the application its correct environment, NOT `process.env` like you would assume.
> 
> ![image](https://private-user-images.githubusercontent.com/15952792/382437084-12530c31-301b-4e7e-99d1-6121ae25be6f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0Njg5ODAsIm5iZiI6MTczNDQ2ODY4MCwicGF0aCI6Ii8xNTk1Mjc5Mi8zODI0MzcwODQtMTI1MzBjMzEtMzAxYi00ZTdlLTk5ZDEtNjEyMWFlMjViZTZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE3VDIwNTEyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3NzBkMDI0OTAzNTRmYmExZGFiZjJhN2ZmZDhlOTg5YmUzYWQ1OWQ2NmYzYTUzYjFlZTI3NDQwOGExNzE1ZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KMmbjg4k0x3kl8Qxg7wfi7XIC-rj5qckV7-o7UjN2B4)
> 
> So, in summary, Workers only exposes environment variables per `fetch()` as the passed `env` parameter, which I cannot find how to access via the generated page module.
> 
> I'd love to contribute to fix this if anyone can point me in the right direction.

@nootsponge I think the problem is here https://github.com/withastro/astro/blob/8da231855162af245f2b3664babb68dff0ba390f/packages/db/src/core/integration/vite-plugin-db.ts#L159

Is it possible to had a cloudflare case here ?

**Comment by NotSujanSharma** [comment]
is it solved yet?

**Comment by ionTea** [comment]
I'm having the same issue, anyone found a workaround for this?

**Comment by jonaspm** [comment]
this worked for me inside astro.config.ts file:

![Image](https://github.com/user-attachments/assets/532f33b4-7a3d-41a9-ba83-9f4e01ba6281)

**Comment by alexanderniebuhr** [comment]
I'm going to close this issue since Astro & Cloudflare have introduced updates which should make this work. If you still have any issues, please open a new issue for that.
